src/containers/CheckoutPage/CheckoutPage.js

```diff
@@ -237,17 +237,30 @@ export class CheckoutPageComponent extends Component {
     this.setState({ submitting: true });
 
     const { history, speculatedTransaction, currentUser, paymentIntent, dispatch } = this.props;
-    const { stripe, card, message } = values;
+    const { stripe, card, message, formId, formValues } = values;
+    const { name } = formValues;
+    const { addressLine1, addressLine2, postalCode, city, state, country } = formValues[formId];
+
+    const billingDetails = {
+      cardholderName: name,
+      billingAddress: {
```

Comment:
Should we just use `address` here?
https://stripe.com/docs/api/charges/object#charge_object-billing_details

Reply:
```js
// Refactoring: I'd name the field as "cardholderName"
const billingDetails = {
    name: cardholderName,
    address: {
      city,
      country,
      line1: addressLine1,
      line2: addressLine2,
      postal_code: postalCode,
      state,
    },
    email: ensureCurrentUser(currentUser).attributes.email,
};
```

Then you could already fill up the `paymentParams` constant in line 178.
```js
billing_details: billingDetails,
```

Reply:
Edit. Sorry, I gave a bad suggestion: name attribute must be "name" not "cardholderName"
https://developers.google.com/web/fundamentals/design-and-ux/input/forms/